crates/tests/src/integration/masp.rs

```diff
@@ -1442,6 +1442,1069 @@ fn enable_rewards_after_shielding() -> Result<()> {
     Ok(())
 }
 
+/// In this test we verify that the results of auto-compounding are
+/// approximately equal to what is obtained by manually unshielding and
+/// reshielding each time.
+#[test]
+fn auto_compounding() -> Result<()> {
+    // This address doesn't matter for tests. But an argument is required.
+    let validator_one_rpc = "http://127.0.0.1:26567";
+    // Download the shielded pool parameters before starting node
+    let _ = FsShieldedUtils::new(PathBuf::new());
+    let (mut node, _services) = setup::setup()?;
+    // Wait till epoch boundary
+    node.next_masp_epoch();
+    // Send 0.1 BTC from Albert to PA
+    let captured = CapturedOutput::of(|| {
+        run(
+            &node,
+            Bin::Client,
+            apply_use_device(vec![
+                "shield",
+                "--source",
+                ALBERT,
+                "--target",
+                AA_PAYMENT_ADDRESS,
+                "--token",
+                BTC,
+                "--amount",
+                "0.1",
+                "--signing-keys",
+                ALBERT_KEY,
+                "--node",
+                validator_one_rpc,
+            ]),
+        )
+    });
+    assert!(captured.result.is_ok());
+    assert!(captured.contains(TX_APPLIED_SUCCESS));
+
+    // Send 0.1 BTC from Albert to PA
```

Comment:
I've forgotten who the `AB_PAYMENT_ADDRESS` belongs to. Is it Albert's diversified address? Either way, a little clarification in the comment here seems apropos.

Reply:
I've clarified the target payment addresses, thanks!